test/models/user_test.rb

```diff
@@ -37,4 +37,293 @@ class UserTest < ActiveSupport::TestCase
+      .any_instance
+      .expects(:get_principal_access)
+      .once
+      .returns(fake_rbac_api_response)
+
+    assert_equal Rbac::GLOBAL_RESOURCE_ACCESS, current_user.inventory_groups
+  end
+
+  test 'global access to hosts' do
+    account = FactoryBot.create(:account)
+    current_user = FactoryBot.create(:user, account: account)
+    fake_rbac_api_response = RBACApiClient::AccessPagination.new(
+      data: [
+        RBACApiClient::Access.new(
+          permission: 'inventory:hosts:*', # inventory:hosts:*
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'groups.id',
+                value: %w[
+                  78e3dc30-cec3-4b49-be2d-37482c74a9ad
+                ],
+                operation: 'in'
+              }
+            }
+          ]
+        )
+      ]
+    )
+    RBACApiClient::AccessApi
+      .any_instance
+      .expects(:get_principal_access)
+      .once
+      .returns(fake_rbac_api_response)
+
+    assert_equal ['78e3dc30-cec3-4b49-be2d-37482c74a9ad'], current_user.inventory_groups
+  end
+
+  test 'ignores unverified keys or permissions' do
+    account = FactoryBot.create(:account)
+    current_user = FactoryBot.create(:user, account: account)
+    fake_rbac_api_response = RBACApiClient::AccessPagination.new(
+      data: [
+        RBACApiClient::Access.new(
+          permission: 'inventory:groups:write', # entry with unverified permission is ignored
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'groups.id',
+                value: %w[
+                  77e3dc30-cec3-4b49-be2d-37482c74a9ac
+                  77e3dc30-cec3-4b49-be2d-37482c74a9ad
+                ],
+                operation: 'in'
+              }
+            }
+          ]
+        ),
+        RBACApiClient::Access.new(
+          permission: Rbac::INVENTORY_HOSTS_READ,
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'foo.id', # entry with unverified key is ignored
+                value: ['77e3dc30-cec3-4b49-be2d-37482c74a9ac'],
+                operation: 'in'
+              }
+            }
+          ]
+        )
+      ]
+    )
+    RBACApiClient::AccessApi
+      .any_instance
+      .expects(:get_principal_access)
+      .once
+      .returns(fake_rbac_api_response)
+
+    assert_equal [], current_user.inventory_groups
+  end
+
+  test 'does not respond to unexpected values' do
+    account = FactoryBot.create(:account)
+    current_user = FactoryBot.create(:user, account: account)
+    fake_rbac_api_response = RBACApiClient::AccessPagination.new(
+      data: [
+        RBACApiClient::Access.new(
+          permission: Rbac::INVENTORY_HOSTS_READ,
+          resource_definitions: [
+            {
+              attributeFilter: {
+                key: 'group.id',
+                value: nil, # nil
+                operation: 'equal'
+              }
```

Comment:
According to the specs, this represents access to only ungrouped hosts. So this test is not according to the specs :disappointed: